// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.messages";
option java_outer_classname = "MessagesProto";
option objc_class_prefix = "HLW";

package messages;

// The greeting service definition.
service FileServer {
  // Sends a greeting
  rpc GetFile (FileRequest) returns (stream FileChunk) {}
}

// The request message containing the user's name.
message FileRequest {
  string filepath = 1;
}

// Response stream from server to client
message FileChunk {
  string filemd5 = 1;  // optional, MD5 for the whole file
  bytes data = 2;
}
